third_party/clutz.bzl

```diff
@@ -49,25 +51,26 @@ def deprecated_extract_dts_from_closure_libraries(ctx):
   js = collect_js(deps, ctx.files._closure_library_base)
   if not js.srcs:
     return None
-  js_typings = ctx.new_file(ctx.bin_dir, "%s-js-typings.d.ts" % ctx.label.name)
+  js_typings = ctx.actions.declare_file(paths.join(
+      ctx.bin_dir.path, "%s-js-typings.d.ts" % ctx.label.name))
   # File.extension does not have leading "." whereas JS_FILE_TYPE does.
   clutz_js_externs = [f for f in ctx.files._clutz_externs
                       if '.%s' % f.extension in JS_FILE_TYPE]
   srcs = depset(transitive=[depset(clutz_js_externs), js.srcs])
   args = ["-o", js_typings.path]
-  for src in srcs:
+  for src in srcs.to_list():
     args.append(src.path)
   if getattr(ctx.attr, "clutz_entry_points", None):
     args.append("--closure_entry_points")
     args.extend(ctx.attr.clutz_entry_points)
-  ctx.action(
-      inputs=list(srcs),
+  ctx.actions.run(
+      inputs=srcs,
```

Comment:
Of course no action required: 
It looks like `inputs` can be a `sequence` (does it mean `list`?) too but is there a reason why we are preferring depset over it? https://docs.bazel.build/versions/master/skylark/lib/actions.html#run

Reply:
Yes, `inputs` can be a list, and indeed the mechanical change here would
have been to change `list(srcs)` to `srcs.to_list()`, but `to_list()`
can be expensive and there’s no reason that we actually need to call it.